packages/video_player/lib/video_player.dart

```diff
@@ -148,14 +150,16 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   /// package and null otherwise.
   VideoPlayerController.asset(this.dataSource, {this.package})
       : dataSourceType = DataSourceType.asset,
+        formatHint = null,
         super(VideoPlayerValue(duration: null));
 
   /// Constructs a [VideoPlayerController] playing a video from obtained from
   /// the network.
   ///
   /// The URI for the video is given by the [dataSource] argument and must not be
-  /// null.
-  VideoPlayerController.network(this.dataSource)
+  /// null. The [formatHint] option allows the caller to specify the video
+  /// format in the case where it cannot be detected via url (Android only).
```

Comment:
I think it is the opposite based on the JAVA implementation. It seems that the formatHint will override the url detection. Maybe:
`Android only: The [formatHint] option allows the caller to override the video format from url`?